aiida/orm/nodes/data/jsonable.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+"""Data plugin that allows to easily wrap objects that are JSON-able."""
+import importlib
+import json
+import typing
+
+from aiida.orm import Data
+
+try:
+    from typing import Protocol
+except ImportError:  # Python <3.8 doesn't have `Protocol` in the stdlib
+    from typing_extensions import Protocol  # type: ignore[misc]
+
+
+class JsonSerializableProtocol(Protocol):
+
+    def as_dict(self) -> typing.MutableMapping[typing.Any, typing.Any]:
```

Comment:
nitpick: I think to be jsonable, keys need to strings no?

```suggestion
    def as_dict(self) -> typing.MutableMapping[str, typing.Any]:
```

Reply:
meh, I guess it would fail with monty's less defined type: https://github.com/materialsvirtuallab/monty/blob/6149bcd9c3b30fa35ad6dbad608a130b7860fc72/monty/json.py#L115, so don't worry about it